src/Workspaces/Core/Portable/Utilities/IDictionaryExtensions.cs

```diff
@@ -42,6 +43,16 @@ internal static class IDictionaryExtensions
             collection.Add(value);
         }
 
+        public static void MultiAdd<TKey, TValue>(this IDictionary<TKey, ImmutableArray<TValue>> dictionary, TKey key, TValue value, ImmutableArray<TValue> defaultArray)
+        {
+            if (!dictionary.TryGetValue(key, out var collection))
+            {
+                collection = ImmutableArray<TValue>.Empty;
+            }
+
+            dictionary[key] = collection.IsEmpty && value == default ? defaultArray : collection.Add(value);
```

Comment:
```suggestion
            dictionary[key] = collection.IsEmpty && value == defaultArray[0] ? defaultArray : collection.Add(value);
```

@panopticoncentral One small last suggestion (sorry) but hopefully you can just hit apply: this is mostly to guard against if somebody handed in an array that *wasn't* filled with a default element, we'd still be functionally correct. It also means that if we ever have that case down the road it'll work too if the non-default value is common.